content/client-lib-development-guide/features.textile

```diff
@@ -368,9 +368,8 @@ h3(#realtime-connection). Connection
 ** @(RTN9a)@ Is @null@ until connected
 ** @(RTN9b)@ Is a unique private connection key provided by Ably that is used to reconnect and retain connection state following an unexpected disconnection. You should have a test to ensure multiple connected clients have unique connection keys
 * @(RTN10)@ @Connection#serial@ attribute:
-** @(RTN10a)@ Is set to -1 once @CONNECTED@
-** @(RTN10b)@ Is always 0 when the first message has been sent and an @ACK@ has been received confirming receipt of the message. The @serial@ is updated whenever a @ProtocolMessage@ is received that contains a @connectionSerial@ value. A test should exist that checks that the serial number is not updated when a message is sent, but then increments by one when the @ACK@ is received
-** @(RTN10c)@ The last known @ProtocolMessage@ @connectionSerial@ is used when restoring connection state
+** @(RTN10a)@ Is @null@ until connected, and is then set from the @connectionSerial@ attribute of the @CONNECTED@ @ProtocolMessage@
```

Comment:
sure, can change to that. (will need to make the same change in a few other places for consistency, e.g. connection.id, connection.key, channel.attachSerial).

Maybe we should just have an interpretation section with things like "References to a value being null should be interpreted as the language-idiomatic way of indicating an unset value", clarifications that 'returning an error' can include calling back with it or raising an exception per language conventions, etc

Reply:
I agree that some general wording will help.

> 'returning an error' can include calling back with it or raising an exception per language conventions, etc

Yes, and for situations like this I've tended to use expressions such as "an error is indicated" or "results in an error" rather than explicitly say "return".

For the `connectionSerial` case I think it's fine to say that the value is initially unset and remains unset until .....